account/bank_account.py

```diff
@@ -14,17 +13,17 @@ def generate_account_id(self):
     def to_dict(self):
         return {
             "user_id": self.user_id,
-            "full_name": self.full_name,
+            "full_name": self.account_type,
             "balance": self.balance,
             "account_id": self.account_id
         }
 
     @classmethod
-    def from_dict(cls, data:UserDict):
+    def from_dict(cls, data):
         account = cls(
             user_id=data["user_id"],
-            full_name=data.get("full_name", ""),
-            balance=data.get("balance", 0)
-        )
+            account_type=data["full_name"],
+            balance=data["balance"]
```

Comment:
`data.get("full_name", "")` is better than `data["full_name"]` because the latter causes exception when the property does not exist.